### PR TITLE
Nzpmc 047/mobile support buttons

### PR DESCRIPTION
**Pull request by matthewaptaylor**
**Issue**

Fixes mobile support issues and login bug

**Solution**

Next button will no longer go out of frame on a mobile device, menu closes when clicked.

Whenever a token will expire within 10 seconds, or has expired, and a request to GraphQL is made, Apollo will first check if a new token exists, and if it does, put it into local storage.

**Risk**

None known

### CODE REVIEW

**Comment by danieltorrey** [comment]
LGTM

**Review by danieltorrey — APPROVED** [review]
LGTM

File: src/vue-apollo.js [hunk 1]
```diff
@@ -47,7 +48,27 @@ const defaultOptions = {
...
     // Override the way the Authorization header is set
-    // getAuth: (tokenName) => ...
+    getAuth: async (tokenName) => {
+        // Check if auth token in local storage is nearing expiration
+        const jwt = localStorage[tokenName]
+        const payload = jwt.split('.')[1]
+        const exp = JSON.parse(atob(payload)).exp
+        const now = new Date().valueOf() / 1000
+
+        console.log(now - exp, now, exp)
```
Comment (Will-ZYS) [inline src/vue-apollo.js:58]: Remove console.log?

File: src/vue-apollo.js [hunk 2]
```diff
@@ -47,7 +48,27 @@ const defaultOptions = {
     // cache: myCache
 
     // Override the way the Authorization header is set
-    // getAuth: (tokenName) => ...
+    getAuth: async (tokenName) => {
```
Comment (Will-ZYS) [inline src/vue-apollo.js:51]: Interesting solution

**Review by Will-ZYS — APPROVED** [review]
LGTM

**Pull request merged**